packages/hash/api/src/model/page.model.ts

```diff
@@ -147,9 +151,11 @@ class __Page extends Entity {
       entityId,
     });
 
-    /** @todo: ensure this is entity has the page system type */
+    if (dbPage?.entityTypeId !== (await this.getEntityType(client)).entityId) {
+      return null;
+    }
```

Comment:
**Blocking:** We should throw an error instead, `null` suggests the entity doesn't exist at all. Also when accessing static methods can we use the name of the class instead of `this` to call the method, so that it becomes more obvious where the method was defined? (i.e. `Page.getEntityType` instead of `this.getEntityType`)
```suggestion
    if (dbPage) {
      if (dbPage.entityTypeId !== (await Page.getEntityType(client)).entityId) {
        throw new Error(
          `Entity with entityId ${entityId} in account ${accountId} is not a Page`,
        );
      }

      return new Page(dbPage);
    }

    return null;
```

Reply:
Yes agreed. The reason why I ended up returning null was that code references already were doing a null check and throwning. I'll consolidate those